x/nodes/keeper/valStateChanges_test.go

```diff
@@ -498,6 +493,97 @@ func TestValidatorStateChange_OutputAddressEdit(t *testing.T) {
 	assert.Equal(t, validatorCur.OutputAddress, outputAddress)
 }
 
+func TestValidatorStateChange_Delegators(t *testing.T) {
+	ctx, _, k := createTestInput(t, true)
+
+	originalUpgradeHeight := codec.UpgradeHeight
+	originalTestMode := codec.TestMode
+	originalNCUST := codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey]
+	originalOEDIT := codec.UpgradeFeatureMap[codec.OutputAddressEditKey]
+	originalReward := codec.UpgradeFeatureMap[codec.RewardsDelegatorKey]
+	t.Cleanup(func() {
+		codec.UpgradeHeight = originalUpgradeHeight
+		codec.TestMode = originalTestMode
+		codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey] = originalNCUST
+		codec.UpgradeFeatureMap[codec.OutputAddressEditKey] = originalOEDIT
+		codec.UpgradeFeatureMap[codec.RewardsDelegatorKey] = originalReward
+	})
+
+	// Enable EditStake, NCUST, and OEDIT
+	codec.TestMode = 0
+	codec.UpgradeHeight = -1
+	codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey] = -1
+	codec.UpgradeFeatureMap[codec.OutputAddressEditKey] = -1
+
+	// Prepare accounts
+	outputPubKey := getRandomPubKey()
+	operatorPubKey1 := getRandomPubKey()
+	operatorPubKey2 := getRandomPubKey()
+	operatorAddr1 := sdk.Address(operatorPubKey1.Address())
+	outputAddress := sdk.Address(outputPubKey.Address())
+	operatorAddr2 := sdk.Address(operatorPubKey2.Address())
+
+	// Fund output address for two nodes
+	stakeAmount := sdk.NewCoin(k.StakeDenom(ctx), sdk.NewInt(k.MinimumStake(ctx)))
+	assert.Nil(t, fundAccount(ctx, k, outputAddress, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, outputAddress, stakeAmount))
+
+	runStake := func(
+		operatorPubkey crypto.PublicKey,
+		delegators map[string]uint32,
+		signer crypto.PublicKey,
+	) sdk.Error {
+		msgStake := types.MsgStake{
+			Chains:     []string{"0021", "0040"},
+			ServiceUrl: "https://www.pokt.network:443",
+			Value:      stakeAmount.Amount,
+			PublicKey:  operatorPubkey,
+			Output:     outputAddress,
+			Delegators: delegators,
+		}
+		return handleStakeForTesting(ctx, k, msgStake, signer)
+	}
+
+	singleDelegator := map[string]uint32{}
+	singleDelegator[getRandomValidatorAddress().String()] = 1
+
+	// Attempt to set a delegators before the upgrade --> The field is ignored
+	assert.Nil(t, runStake(operatorPubKey1, singleDelegator, outputPubKey))
+	validatorCur, found := k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.Delegators)
+
+	// Enable DELE
+	codec.UpgradeFeatureMap[codec.RewardsDelegatorKey] = -1
+
+	// Attempt to change the delegators with output's signature --> Fail
```

Comment:
Was this part of the proposal passed? Output can't change delegators but the operator can?

Reply:
Yes.  See this https://forum.pokt.network/t/pip32-unleashing-the-potential-of-non-custodial-node-running/4796.

> Another topic we need to discuss is “editing”. The upgrade to RC-0.10.4 enabled the new feature Output Address Edit [3], which allows an MsgStake transaction signed by the current output address to change the output address to another. Now who should be able to change delegators? We propose that only the owner of the node operator should have the authority to make changes as follows:

Reply:
Can you #PUC with the link?